src/jwks-fetch.js

```diff
@@ -1,54 +1,67 @@
 'use strict'
 
 const fetch = require('node-fetch')
-const MISSING_KEY_ERROR = 'No matching key found in the set.'
-const NO_KEYS_ERROR = 'No keys found in the set.'
+const lru = require('tiny-lru')
 
-async function jwksFetch ({ domain, alg, kid, cache = null }) {
-  const cacheKey = `${alg}:${kid}:${domain}`
+const MISSING_KEY_ERROR = 'No matching key found in the set.';
 
-  const cached = cache ? cache.get(cacheKey) : null
+async function buildJwksFetch (cacheProps = { max: 100, ttl: 60 * 1000 }) {
```

Comment:
why is this an async function?

Note that the default values do not work the way you think they do, e.g. `buildJwksFetch({})` will have `max` as `undefined`.

Reply:
Ah I see, thanks!